cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -228,8 +228,25 @@ public void eventReceived(Action action, KafkaRebalance kafkaRebalance) {
                             kafkaRebalance.getStatus() != null ? rebalanceStateConditionType(kafkaRebalance.getStatus()) : null,
                             ANNO_STRIMZI_IO_REBALANCE, rawRebalanceAnnotation(kafkaRebalance));
 
-                    withLock(reconciliation, LOCK_TIMEOUT_MS,
-                        () -> reconcileRebalance(reconciliation, action == Action.DELETED ? null : kafkaRebalance));
+                    if (action == Action.MODIFIED) {
+                        KafkaRebalanceBuilder patchedKafkaRebalance = new KafkaRebalanceBuilder(kafkaRebalance);
+
+                        if (patchedKafkaRebalance.buildStatus().getObservedGeneration() != (kafkaRebalance.getMetadata().getGeneration())) {
```

Comment:
```suggestion
                        if (patchedKafkaRebalance.buildStatus().getObservedGeneration() != kafkaRebalance.getMetadata().getGeneration()) {
```

Also, what about null checks? How do you know that the status cannot be null?